### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by mishagitcode**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(error) for error in errors],
+        "path": file_path,
+        "status": "failed"
```
Comment (EdAlekseiev) [inline app/main.py]: If `errors` is empty list status must be `passed `

File: app/main.py [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        format_single_linter_file(key, value) if value != [] else {
+            "errors": value,
+            "path": key,
+            "status": "passed"
+        }
```
Comment (EdAlekseiev) [inline app/main.py]:
```suggestion
        format_single_linter_file(key, value)
```

**Review by EdAlekseiev — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -1,13 +1,35 @@
...
-    pass
+    return {
+        "errors": [format_linter_error(error) for error in errors],
+        "path": file_path,
+        "status": "failed"
+    } if errors != [] else {
+        "errors": [],
+        "path": file_path,
+        "status": "passed"
+    }
```
Comment (Bohdan-Salamakha) [inline app/main.py]:
```suggestion
    return {
        "errors": [format_linter_error(error) for error in errors],
        "path": file_path,
        "status": "failed" if errors else "passed"
    }
```

File: app/main.py [hunk 4]
```diff
@@ -1,13 +1,35 @@
...
+
+            "errors": value,
+
+            "path": key,
+
+            "status": "passed"
+
+        }
+        for key, value in linter_report.items()
+    ]
```
Comment (Bohdan-Salamakha) [inline app/main.py]:
```suggestion
    return [
        format_single_linter_file(file_path, error)
        for file_path, error in linter_report.items()
    ]
```

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]

**Review by Bohdan-Salamakha — APPROVED** [review]

**Pull request closed without merging**